src/main/scala/io/iohk/ethereum/db/dataSource/DataSource.scala

```diff
@@ -0,0 +1,26 @@
+package io.iohk.ethereum.db.dataSource
+
+trait DataSource {
+
+  type Key = Array[Byte]
+  type Value = Array[Byte]
+
+  /**
+    * This function obtains the associated value to a key, if there exists one.
+    *
+    * @param namespace which will be searched for the key.
```

Comment:
> IODB requires all the keys to be of the same size

I'd take care of this contraint in `IodbDataSource`. Apart from that `Key`, being an `IndexedSeq[Byte]` does not enforce fixed length of its values.